ema/src/Ema/Server.hs

```diff
@@ -95,56 +93,44 @@ runServerWithWebSocketHotReload host mport model = do
             log LevelInfo "Connected"
             let askClientForRoute = do
                   msg :: Text <- liftIO $ WS.receiveData conn
-                  -- TODO: Let non-html routes pass through.
-                  let pathInfo = pathInfoFromWsMsg msg
-                  log LevelDebug $ "<~~ " <> show pathInfo
-                  pure pathInfo
-                decodeRouteWithCurrentModel pathInfo = do
-                  val <- LVar.get model
-                  pure $ routeFromPathInfo val pathInfo
-                sendRouteHtmlToClient pathInfo s = do
-                  decodeRouteWithCurrentModel pathInfo >>= \case
+                  log LevelDebug $ "<~~ " <> show msg
+                  pure msg
+                sendRouteHtmlToClient path m =
+                  case decodeUrlRoute m path of
                     Left err -> do
                       log LevelError $ badRouteEncodingMsg err
                       liftIO $ WS.sendTextData conn $ emaErrorHtmlResponse $ badRouteEncodingMsg err
                     Right Nothing ->
                       liftIO $ WS.sendTextData conn $ emaErrorHtmlResponse decodeRouteNothingMsg
                     Right (Just r) -> do
-                      renderCatchingErrors s r >>= \case
+                      renderCatchingErrors m r >>= \case
                         AssetGenerated Html html ->
                           liftIO $ WS.sendTextData conn $ html <> toLazy wsClientHtml
                         -- HACK: We expect the websocket client should check for REDIRECT prefix.
                         -- Not bothering with JSON response to avoid having to JSON parse every HTML dump.
                         AssetStatic _staticPath ->
-                          liftIO $ WS.sendTextData conn $ "REDIRECT " <> toText (review (fromPrism_ $ enc s) r)
+                          liftIO $ WS.sendTextData conn $ "REDIRECT " <> toText (review (fromPrism_ $ routePrism m) r)
```

Comment:
@lucasvreis What is the reason for adjacent changes like this?

- Replace `s` with `m`
- Replace `enc` with `routePrism`

I am just trying to narrow down to the "Simplifies the websocket loop logic" part, which is what I mostly care about.

Reply:
I think a PR that does just the "Simplifies the websocket loop logic" part, but without the refactoring, would be most helpful.

Reply:
I assume that the main impact of this PR is that it fixes the double-fetch, which alone would be cause for making another release of Ema.

Reply:
> What is the reason for adjacent changes like this?

zero particular reasons. I just did git amend and undid some of the changes of the second commit until it looked reasonable. I was not caring about minimizing diffs or anything, so those lines were already shadowed by the indentation change.

Reply:
> I think a PR that does just the "Simplifies the websocket loop logic from ... to ..." part, but without the refactoring, would be most helpful.

Understandable, I will reopen with more things split.